staging/src/k8s.io/sample-apiserver/pkg/admission/wardleinitializer/wardleinitializer.go

```diff
@@ -18,26 +18,33 @@ package wardleinitializer
 
 import (
 	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/server"
+	clientset "k8s.io/sample-apiserver/pkg/client/clientset/internalversion"
 	informers "k8s.io/sample-apiserver/pkg/client/informers/internalversion"
 )
 
 type pluginInitializer struct {
-	informers informers.SharedInformerFactory
+	config *server.Config
 }
 
 var _ admission.PluginInitializer = pluginInitializer{}
 
 // New creates an instance of wardle admission plugins initializer.
-func New(informers informers.SharedInformerFactory) (pluginInitializer, error) {
+func New(c *server.Config) (pluginInitializer, error) {
 	return pluginInitializer{
-		informers: informers,
+		config: c,
 	}, nil
 }
 
 // Initialize checks the initialization interfaces implemented by a plugin
 // and provide the appropriate initialization data
 func (i pluginInitializer) Initialize(plugin admission.Interface) {
+	client, err := clientset.NewForConfig(i.config.LoopbackClientConfig)
+	if err != nil {
+		panic(err)
+	}
+	informerFactory := informers.NewSharedInformerFactory(client, i.config.LoopbackClientConfig.Timeout)
```

Comment:
This cannot live here. The shared informer factory must be shared. It cannot be per admission plugin. With https://github.com/kubernetes/kubernetes/pull/56660/files#r156612725 in mind, we have to move this informer factory into the WardleServer config and pass that one into the initializer above. Then we have access here to it.

Still wondering, how we can elegantly detangle this in a cleaner way. But gut says that we are still not there yet.

Reply:
Thanks for point out this though this is sample-apiserver. It should make use of shared informer by a shared way.  I will think a lot more.

Reply:
Just read code again, this is a plugin initializer. It is used to initialize other admission plugin. Is that not shared by all admission plugins?

Reply:
BTW, no two plugin initializers will initialize plugins twice by `wants.SetInternalKubeInformerFactory(i.informers)`.

Reply:
it's shared. But the shared informer should be shared process-wide, not only among admission plugins. It's just the wrong place to create those objects. Admission plugin are only one consumer among many.

Reply:
this might be ok in this special case, but sample-apiserver is supposed to be a blueprint for more complex apiservers. We don't want to promotoe bad practices.

Reply:
oh, I know. We need more thought later.